### PR TITLE
vacancies in teams

### LINKED ISSUE #456: vacancies in teams
1. Create a vacancy model which has links to a group and optionally a user. Also should have a link to a hire stream from taskany/hire
2. Show it in the group list - the empty ones
3. Allow creation via 'create' menu
4. Group link is mandatory
5. Allow deletion of a vacancy - when deleted from a group, it should become 'archived' and not be listed

### PR DESCRIPTION
**Pull request by nyet-ty**
Resolve #456 

## PR includes

-   [x] Feature

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -19,4 +19,8 @@ export const config = {
         apiUrlUpdate: process.env.EXTERNAL_SERVICE_UPDATE_PROFILE_API_URL,
         apiToken: process.env.EXTERNAL_SERVICE_API_TOKEN,
     },
+    hireIntegtation: {
```
Comment (tsumo) [inline src/config.ts:22]: typo 'integ_t_ation'

File: src/components/HireStreamComboBox/HireStreamComboBox.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    onChange: (hireStream: Nullish<HireStream>) => void;
+}
+
+export const HireStreamComboBox = ({ hireStream, onChange }: HireStreamComboBoxProps) => {
+    const [search, setSearch] = useState('');
+    const suggestionsVisibility = useBoolean(false);
+    const [selectedHireStream, setSelectedHireStream] = useState(hireStream);
+    const hireStreamQuery = trpc.hireIntegration.getHireStreamList.useQuery();
+
+    const items = hireStreamQuery.data?.filter(({ name }) => name.toLowerCase().includes(search.toLowerCase()));
```
Comment (tsumo) [inline src/components/HireStreamComboBox/HireStreamComboBox.tsx:24]: memo maybe

File: src/modules/hireIntegrationMethods.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { config } from '../config';
+
+import { HireStream } from './hireIntegrationTypes';
+
+export const hireIntegrationMethods = {
+    getHireStreams: async () => {
+        if (!config.hireIntegtation.apiToken) throw new Error();
```
Comment (tsumo) [inline src/modules/hireIntegrationMethods.ts:7]: Error text

File: src/modules/vacancyHooks.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    });
+
+    const archiveVacancy = trpc.group.archive.useMutation({
+        onSuccess: () => {
+            utils.vacancy.invalidate();
+        },
+    });
+
+    return {
+        createVacancy: (data: CreateVacancy) => notifyPromise(createVacancy.mutateAsync(data), 'groupCreate'),
```
Comment (tsumo) [inline src/modules/vacancyHooks.ts:28]: vacancyCreate

File: src/modules/vacancySchemas.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    hireStream: z.string().optional(),
+});
+export type GetVacancyList = z.infer<typeof getVacancyListSchema>;
+
+export const editVacancySchema = z.object({
+    id: z.string(),
+    name: z.string().optional(),
+    groupId: z.string().optional(),
+    hireStream: z.string().optional(),
+    archived: z.boolean().optional(),
```
Comment (tsumo) [inline src/modules/vacancySchemas.ts:37]: Let's leave archived status change only in separate endpoints. It often requires a lot of additional logic

File: src/modules/vacancyTypes.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+export interface UserSettings {
+    userId: string;
+    theme: Theme;
+}
+
+export type VacancyWithGroup = Vacancy & {
+    group: Group;
+};
+
+export type VacancyWithGroupAndUser = VacancyWithGroup & {
```
Comment (tsumo) [inline src/modules/vacancyTypes.ts:15]: This kind of naming of types was out of control in hire. Let's try do compound types instead, like in userTypes.tsx - like `User & UserSupervisor`. This way every component can only ask for things that it really needs

File: src/trpc/router/hireIntegrationRouter.ts [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import { z } from 'zod';
+
+import { router, protectedProcedure } from '../trpcBackend';
+import { hireIntegrationMethods } from '../../modules/hireIntegrationMethods';
+
+export const hireIntegrationRouter = router({
+    getHireStreamList: protectedProcedure.input(z.void()).query(() => {
```
Comment (tsumo) [inline src/trpc/router/hireIntegrationRouter.ts:7]: don't need to specify input with void for non-rest endpoints

**Review by tsumo — CHANGES_REQUESTED** [review]

**Review by 9teen90nine — APPROVED** [review]

File: src/components/TeamVacancies/TeamVacancies.tsx [hunk 8]
```diff
@@ -0,0 +1,33 @@
+import styled from 'styled-components';
+import { gapM, gapS } from '@taskany/colors';
+
+import { NarrowSection } from '../NarrowSection';
+import { trpc } from '../../trpc/trpcClient';
+import { VacancyListItemEditable } from '../VacancyListItemEditable';
+
+import { tr } from './TeamVacancies.i18n';
+
+const StyledUserList = styled.div`
```
Comment (tsumo) [inline src/components/TeamVacancies/TeamVacancies.tsx:10]: VacancyList

File: src/components/VacancyListItem.tsx [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+    align-items: center;
+    flex-wrap: nowrap;
+`;
+
+export const VacancyListItem = ({ vacancy }: VacancyListItemProps) => {
+    return (
+        <StyledWrapper>
+            <IconSearchOutline size="s" color={gray9} />
+            <Text>
+                <Link>{vacancy.name}</Link>
```
Comment (tsumo) [inline src/components/VacancyListItem.tsx:25]: Not a link

**Review by tsumo — CHANGES_REQUESTED** [review]

**Review by tsumo — APPROVED** [review]

**Pull request merged**